tests/Artifact.py

```diff
@@ -19,6 +19,14 @@ def testGetArtifactsFromWorkflow(self):
             repr(artifact), 'Artifact(name="vscode-codeql-extension", id=10495898)'
         )
 
+    def testGetArtifactsFromWorkflowWithName(self):
```

Comment:
Your test name suggests this gets artifacts for a workflow. Given the name, we would expect only a single artifact. That is the reason for my confusion. Test should be renamed:

```suggestion
    def testGetArtifactsFromRepoWithName(self):
```

Reply:
Would you consider to add the name filter to `Repository.get_workflow_run(...).get_artifacts` as well?

Reply:
Maybe later in another PR, not in this one.

Reply:
after pyi are merged back to py